File: 724-find-pivot-index/find-pivot-index.go
func pivotIndex(nums []int) int {

    totalsum := 0

    prefixsum := 0

    for _, num := range nums {
        totalsum +=num
    }

    for idx, num := range nums {
        right := totalsum - prefixsum -num 

        if prefixsum == right {
            return idx
        }

        prefixsum += num
    }

    return -1;
    
}

File: 724-find-pivot-index/find-pivot-index.py
class Solution:

    def pivotIndex(self, nums: List[int]) -> int:
        total = sum(nums)
        left = 0

        for i, num in enumerate(nums):
            # total - left - num gives the sum of elements to the right of index i
            if left == total - left - num:
                return i
            left += num

        return -1